src/bsonc.rs

```diff
@@ -28,10 +28,7 @@ impl Bsonc {
         try!(bson::encode_document(&mut buffer, document));
 
         let inner = unsafe {
-            bindings::bson_new_from_data(
-                buffer[..].as_ptr(),
-                buffer.len() as u64
-            )
+            bindings::bson_new_from_data(buffer[..].as_ptr(), (buffer.len() as u32).into())
```

Comment:
Why this change?

Reply:
em....there is a problem with this change, close the pr first..